packages/react-core/src/components/Select/Select.tsx

```diff
@@ -16,10 +35,16 @@ export interface SelectProps extends MenuProps, OUIAProps {
   /** Renderer for a custom select toggle. Forwards a ref to the toggle. */
   toggle: (toggleRef: React.RefObject<any>) => React.ReactNode;
   /** Function callback when user selects an option. */
-  onSelect?: (event?: React.MouseEvent<Element, MouseEvent>, itemId?: string | number) => void;
+  onSelect?: (
+    event?: React.MouseEvent<Element, MouseEvent>,
+    itemId?: string | number,
+    toggleRef?: React.RefObject<any>
```

Comment:
So the whole purpose of having the `toggle` render is so that consumer does not have to pass a ref.  I think we should not expect the consumer to use the ref in this way.  I would opt for the `shouldCloseOnSelect` prop instead.

Reply:
The ref looks like it was a holdover from when we manually wired everything up and wasn't currently being used so I removed them.

I added a `shouldFocusToggleOnSelect` prop (the ref is only for that purpose as opening/closing is entirely user-side). However, the "view more" example will break with this change because it should not be focusing the toggle when the selection is the loading option, so I've left the optional property in for now and added comments to the specific example and description of the callback. Will that work? 

Or alternately we could add an optional `toggleRef` parameter (similar to how `ref` is optional and without anything passed it will default to an internal ref) and let `toggle` accept a node as well as a function to allow a user to control the toggleRef completely. What do you think?

Reply:
@thatblindgeye @nicolethoen which approach do you think is more intuitive?   I m leaning towards the first approach but i can see benefits of the second as well.

Reply:
I'd agree that either approach could work. I might lean more towards the toggleRef prop approach personally as it offers that flexibility and isn't tied to just onSelect being called, but considering we're mainly trying to focus the toggle "on select" I'd be fine with the PR as it is.

Just a couple of things:

- Will this require Dropdown to be updated similarly?
- For the prop description on onSelect and shouldFocusToggleOnSelect, maybe adding in some verbiage to explain that one or the other should be used and not both.

Reply:
Pushed an update with the alternate method, which I can back out if necessary. I kind of like the fallback being manual control of the toggleRef instead of the callback because then a user can control focus in other circumstances as well, not just selection.

Also updated Dropdown similarly.